scripts/download_technique_animations.py

```diff
@@ -8,67 +8,126 @@
 
 import requests
 from PIL import Image
 from lxml import html
 
 FRAME_SIZE = 64
-WIKI_URL = "https://wiki.tuxemon.org/index.php?title=Category:Technique_Animation"
+WIKI_URL = "https://wiki.tuxemon.org"
+
+TUXEMON_ROOT_DIR = pathlib.Path(__file__).resolve().parent.parent
+ANIMATION_DIR = TUXEMON_ROOT_DIR.joinpath(pathlib.Path("mods/tuxemon/animations/technique"))
+CREDITS_TEMPLATE = '* ["{animation_name}"]({animation_url}) {credits_text}'
+CREDITS_FILENAME = "TECHNIQUE_ANIMATION_CREDITS.md"
+
+print("Tuxemon project root dir:", TUXEMON_ROOT_DIR)
+print("Technique animation dir:", ANIMATION_DIR)
 
-# TODO: Is the return here needed or should be boolean to indicate success/fail?
-def download_bytes(url: str, filename: str) -> str:
-    """Downloads a stream of bytes from a given URL to a file defined by 'filename'."""
+
+def download_bytes(url: str, filepath: str) -> None:
+    """Downloads a stream of bytes from a given URL to a file path."""
     req = requests.get(url)
-    with open(filename, "wb") as fp:
+    with open(filepath, "wb") as fp:
         for chunk in req.iter_content(chunk_size=1024):
             if chunk:
                 fp.write(chunk)
-    return filename
 
 
-def process_filename(filename: str) -> str:
+def process_filename(filepath: str) -> str:
     """Extract base filename from an animation file path."""
-    cleaned_filename = os.path.splitext(filename)[0]
+    cleaned_filename = os.path.splitext(os.path.basename(filepath))[0]
     cleaned_filename = cleaned_filename.strip("0123456789_")
     cleaned_filename = cleaned_filename.lower()
     return cleaned_filename
 
 
-def process_gif(filename: str) -> None:
+def process_animation_name(filename: str) -> str:
+    """Convert filename of format 'file_name' to a capitalized 'File Name' animation name/title."""
+    animation_name = " ".join([name_part.capitalize() for name_part in filename.split("_")])
+    return animation_name
+
+
+def download_animation_credits(gif_page_url: str) -> str:
+    """Extract credits text from a GIF animation subpage, from the 'Comments' section."""
+    gif_filename: str = gif_page_url.split("File:")[-1]
+    animation_name: str = process_animation_name(process_filename(gif_filename))
+
+    gifpage_source = requests.get(gif_page_url)
+    gifpage_tree = html.fromstring(gifpage_source.content)
+    comment_blocks = gifpage_tree.xpath("//table[@class='wikitable filehistory']//td[@dir='ltr']")
+
+    credits_text =""
+    for comment_row in comment_blocks: 
+        # Some comment entries are empty
+        if len(comment_row) > 0:
+            for comment_entry in comment_row:
+                # Assemble credits and skip the "Category:" blocks   
+                comment_tail: str = comment_entry.tail
+                if comment_tail is not None and not comment_tail.startswith("Category:"):
+                    credits_text += comment_entry.tail.strip()
+
+    credits_text = credits_text or "Unknown animation source/author"
```

Comment:
Some animations don't have a proper credits record, but they do have an author so that code could also be added if needed.

Reply:
Explain

Reply:
For instance: https://wiki.tuxemon.org/File:Appearance.gif
Sanglorian uploaded (?) the gif, but there is no information of the original source anywhere.

Reply:
We should just ask the uploader